### PR TITLE
Add payment_method_types array returned from Stripe payment_intent into WC_Payments_API_Intention object

### LINKED ISSUE #5939: Include `payment_method_types` in `WC_Payments_API_Intention`
### Description
The `payment_method_types` array included in Stripe's `payment_intent` object can give us helpful information to use in order processing.

https://stripe.com/docs/api/payment_intents/object#payment_intent_object-payment_method_types

### Acceptance criteria

* Update the `WC_Payments_API_Intention` to include the ability to hold  the `payment_method_types` array.
* Update all needed references that create `WC_Payments_API_Intention` objects to include the `payment_method_types`, where applicable.

### Additional notes
This will be helpful with resolving: #5878

### PR DESCRIPTION
**Pull request by jessepearson**
Fixes #5939

#### Changes proposed in this Pull Request

Adds the `payment_method_types` array returned in the Stripe `payment_intent` into the `WC_Payments_API_Intention` object. This will allow this data to be used elsewhere.

#### Testing instructions

* Tests should pass.
* No functionality was changed, it's just additional data available.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `5940` or branch name `add/5939-add-payment-method-types-in-wc-payments-api-intention` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=add/5939-add-payment-method-types-in-wc-payments-api-intention) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 66bea05aaa3a22449f197999d1e867bbfb93bccf
- Build time: 2023-04-03 12:58:21 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by jessepearson** [comment]
Code coverage test is failing with this message:
> Total code coverage is 60.00 % which is below the accepted 60%

There are no tests for this class that's been modified, so I believe it's okay if we ignore this. There's additional tests that will be added in #5888

File: includes/wc-payment-api/class-wc-payments-api-client.php [hunk 1]
```diff
@@ -2291,13 +2291,14 @@ public function deserialize_intention_object_from_array( array $intention_array
...
-		$payment_method     = $intention_array['payment_method'] ?? $intention_array['source'] ?? null;
-		$processing         = $intention_array[ Payment_Intent_Status::PROCESSING ] ?? [];
+		$charge_array         = 0 < $intention_array['charges']['total_count'] ? end( $intention_array['charges']['data'] ) : null;
+		$next_action          = ! empty( $intention_array['next_action'] ) ? $intention_array['next_action'] : [];
+		$last_payment_error   = ! empty( $intention_array['last_payment_error'] ) ? $intention_array['last_payment_error'] : [];
+		$metadata             = ! empty( $intention_array['metadata'] ) ? $intention_array['metadata'] : [];
+		$customer             = $intention_array['customer'] ?? $charge_array['customer'] ?? null;
+		$payment_method       = $intention_array['payment_method'] ?? $intention_array['source'] ?? null;
+		$processing           = $intention_array[ Payment_Intent_Status::PROCESSING ] ?? [];
+		$payment_method_types = $intention_array['payment_method_types'] ?? $intention_array['payment_method_types'] ?? [];
```
Comment (tpaksu) [inline includes/wc-payment-api/class-wc-payments-api-client.php]: If my eyes are still good, I think the two are the same?

Comment (dwainm) [reply includes/wc-payment-api/class-wc-payments-api-client.php]: Looks the same to me.

Comment (jessepearson) [reply includes/wc-payment-api/class-wc-payments-api-client.php]: You were right, it was the same. I just fixed this in https://github.com/Automattic/woocommerce-payments/pull/5940/commits/5cd4bf3f45ee89fc5b59ef014a748e7e65fa9dea

**Review by dwainm — APPROVED** [review]
The feature still works for me. Tests pass locally and the code looks good. Nothing that stands out. I've checked to see if there are other places where the `WC_Payments_API_Intention` is initiated, but I've only seen this one.

**Review by eduardoumpierre — APPROVED** [review]
The changes are easy to understand and the tests are passing. LGTM!

**Pull request merged**